### PR TITLE
[fix] Fix true on policy in new FSDP backend

### PR DESCRIPTION
**Pull request by yueming-yuan**
1. Fix true on policy (bug from refactor): add back true on policy's log probs compute path
2. Fix CI: change from 2 GPUs -> 4 GPUs to aovid OOM

### CODE REVIEW

File: miles/utils/ppo_utils.py [hunk 1]
```diff
@@ -681,3 +681,36 @@ def calculate_log_probs_and_entropy(logits, tokens, tp_group, with_entropy: bool
             entropy = logits.new_zeros((0,))
 
     return log_prob, entropy
+
+
+def calculate_log_probs_and_entropy_for_true_on_policy(
```
Comment (fzyzcjy) [inline miles/utils/ppo_utils.py]: hmm wondering whether we can be a bit more unified w/ existing code (which may or may not be possible)

File: miles/backends/training_utils/loss.py [hunk 2]
```diff
@@ -155,13 +156,20 @@ def get_log_probs_and_entropy(
...
+                with_entropy=with_entropy,
+            )
+        else:
+            log_prob, entropy = calculate_log_probs_and_entropy(
+                logits_chunk,
+                tokens_chunk,
+                parallel_state.tp_group,
+                with_entropy=with_entropy,
+                chunk_size=args.log_probs_chunk_size,
+            )
```
Comment (fzyzcjy) [inline miles/backends/training_utils/loss.py]: what about

```suggestion
        log_prob, entropy = calculate_log_probs_and_entropy(
            logits_chunk,
            tokens_chunk,
            parallel_state.tp_group,
            with_entropy=with_entropy,
            chunk_size=args.log_probs_chunk_size,
        )
```

and `if true on policy` inside calculate_log_probs_and_entropy, since it looks a bit more like impl details

**Review by yushengsu-thu — APPROVED** [review]
Sync off-line already
